utils/src/treev2/quickPickWizard/FindByIdQuickPickStep.ts

```diff
@@ -52,12 +53,16 @@ export class FindByIdQuickPickStep<TContext extends types.QuickPickWizardContext
 class FindByIdPickFilter implements PickFilter {
     constructor(private readonly pickOptions: FindByIdQuickPickOptions) { }
 
-    isAncestorPick(node: vscode.TreeItem): boolean {
+    isAncestorPick(node: vscode.TreeItem, item?: { resources?: ApplicationResource[] }): boolean {
         if (!node.collapsibleState) {
             // can't be an indirect pick if it doesn't have children
             return false;
         }
 
+        if (item?.resources) {
+            return item.resources.some((resource) => this.pickOptions.id.startsWith(resource.id));

```

Comment:
We probably want to guard this better. Check if `item.resources` is an array first. And maybe rename the `resources` property to something more obscure so we don't mistakenly do this check on not-`GroupingItem`s.

I considered making `find` a RGs internal method because then we could do it in a safer way. We could do `instanceof GroupingItem`.